test/functional/fixtures/reporter/test.js

```diff
@@ -1133,6 +1133,142 @@ const experimentalDebug = !!process.env.EXPERIMENTAL_DEBUG;
 
     });
 
+    describe('Reporter Hooks', () => {
+        it('Should handle onBeforeWrite hook', () => {
+            function custom () {
+                return {
+                    taskInfo: { prop: 'val' },
```

Comment:
You don't need checking all methods (all method names calculates with the '_getWriteInitiatorEvent' function).
You can leave on one method with all formatting options.